### DISCUSSION TITLE
Solax X1 Mini with Waveshare POE - Spinning wheel of death

### DISCUSSION THREAD

**Discussion by dougle03** [wills106/homeassistant-solax-modbus#958] [Q&A]
Hi all,
Hoping someone can give me some clues.
I've got a Solax X1 Mini G4 and paired to this I have the Waveshare ETH POE adapter.
I can get to the Waveshare's web page and all seems ok.
I've linked up the Waveshare to the inverter using a cut off (But tested) network cable and connected the solid blue wire to the 485-A and the blue/white wire to the 485-B.
I've added the integration via HACS and gone though the setup but it's frozen at the Initialising stage.
I've a Red Power light and a Blue Link light on the waveshare box.

Have I done something wrong? 
The IP for the Waveshare is correct. I've checked the settings using Vircom and all seem to be ok.
Any help would be greatly appreciated.
Thanks

![image](https://github.com/user-attachments/assets/613bf50a-75ba-4276-bf58-3de59d96b756)
![image](https://github.com/user-attachments/assets/37a07609-987c-4924-b45c-2133ba23a2d9)
![image](https://github.com/user-attachments/assets/1d265cbb-38b1-46a2-9226-bec1a0e49629)

**Comment by dougle03** [comment]
Nevermind.
It started working after I reduced the baud rate to 9600.
Might try higher rates now it's ticking in.

**Reply by soppem** [reply]
I have exactly the same configuration and I can't get modbus to work.
